oa2.md

```diff
@@ -0,0 +1,195 @@
+* OpenAlias records are not organized into a dedicated OpenAlias section, which may result in clutter.
+
+## Advantages of OA1
+
+* DNS TXT records with DNSSEC remains one of the simplest and most robust ways to handle these alias lookups.
+* OA1 includes a version number, which means we can upgrade to a new OAv2 without breaking existing v1 records.
+* OA1 works for arbitrary assets, not just BTC or ETH.
+* OA1 is supported by major wallets Electrum, Cake Wallet, the official Monero wallets, Feather Wallet and more.
+
+## Avoiding Conflicting Standards
+
+OA1 and OA2 do not need to be compatible with every other alias standard, but we should avoid conflicting with them where possible. Some notable ones are:
+
+* Ethereum Name Service (ENS)
+* Namecoin
+* Unstoppable Domains
+* https://github.com/bitcoin/bips/pull/1551
+
+## Proposed OA2 Format
+
+OpenAlias v2 introduces new OpenAlias record types.
+
+### `_openalias-metadata`
+
+`_openalias-metadata` contains various items that are useful across different record types. For example, you can specify a name, image link, social profile links, social profile usernames, or other information.
+
+Example name and image for `donate@openalias.org`:
+
+* TXT name: `_openalias-metadata.donate`
+* TXT content: `oa2 name=OpenAlias Project; image=https://openalias.org/image.png`
+
+#### `_openalias-metadata` Key-Value Pairs
+
+* name
+* description
+* checksum
+* image
+* twitter
+* nostr
+* signal
+* telegram
+
+#### `image`
+
+This is an optional link to an image file on the ***same*** domain and subdomain as the TXT record. For example:
+
+> image=https://getmonero.org/press-kit/symbols/monero-symbol-on-white-480.png
+
+Wallet developers MUST prohibit image lookups on another domain or subdomain.
+
+### `_openalias-payment`
+
+`_openalias-payment` is the core TXT record name for communicating payment information.
+
+The TXT record name begins with `_openalias-payment`. Records placed under that name will act as root records. Subdomain records are possible by affixing that subdomain after `_openalias-payment.`. For `example.openalias.org`, the subdomain would be `_openalias-payment.example`.
+
+The TXT content is:
+
+> oa2 [asset_component] address=[address_component];
+
+Records often appear in these two ways:
+
+> oa2 [asset_ticker]:[asset_network] address=[address_type]:[address];
+
+> oa2 [asset_network] address=[address];
+
+#### `_openalias-payment` Key-Value Pairs
+
+* address (required)
+* amount
+* payment_id
+* address_signature
+* checksum
+
+#### `asset_component`
+
+The `asset_component` is either an `asset_ticker` and `asset_network`, or an `asset_network`. A colon separates the `asset_ticker` and `asset_network`.
+
+`asset_network` is required. `asset_ticker` is recommended.
+
+This is a major improvement in OA. Instead of simply specifying `usdt` as the prefix, recordholders can specify the exact network that they wish to receive USDT on, eg: `usdt:eth`.
```

Comment:
You can use CAIP-19, a proposed standard for identifying assets: https://chainagnostic.org/CAIPs/caip-19

(They have other specs that might also be relevant, for example https://chainagnostic.org/CAIPs/caip-10)

Reply:
Cool stuff. One downside is that it might be _too_ verbose. For example, the following is used to identify Bitcoin:

> bip122:000000000019d6689c085ae165831e93/slip44:0

This is relatively complex/long/verbose for this use-case. Perhaps OpenAlias can rely on these CAIP standards as the overarching reference, and we can implement "short names" that map to these, which would then be used? Does CAIP or anyone else have something like this with short/nicknames?

Reply:
@silverpill I included an initial idea of how to create some connections to CAIP-2 and CAIP-19. Please let me know what you think!

https://github.com/openalias/openalias-spec/pull/1/commits/3afc45c3c97ccb4450a9cd0469e17208cd1e0359

Reply:
I think using nicknames is a good idea, but unless there is a technical limitation on the length of asset ID, both nicknames and full CAIP-19 identifiers should be allowed. If a network or an asset doesn't have a well-known nickname yet, people will use CAIP-19.

Also, shouldn't this repo contain a registry of reserved nicknames?

Reply:
One downside of specifically allowing CAIP-19 instead of an OpenAlias nickname is that CAIP is built on other lists. So a Monero wallet would need to know how to interpret `monero:418015bb9ae982a1975da7d79277c270/slip44:128` as XMR on Monero even if the wallet is otherwise oblivious to slip44.

If we instead limited OpenAlias entries to nicknames only, then wallets which use CAIP and slip44, etc. can still map to them easily, but it's not a requirement. Yes, wallets already compatible with CAIP will need to do another mapping from nickname versus using CAIP directly, which is a minor downside.

Yes, the registry for reserved nicknames will need to be established. OA1 did not establish this, so it needs to be established for OA2 if we don't use CAIP directly.

Reply:
@silverpill it would be excellent if you could join us on Matrix or IRC (links at the top of this document). I want to pick your brain about a few more things, since it seems like you have a lot of familiarity with CAIP.